src/webgl/p5.Framebuffer.js

```diff
@@ -0,0 +1,552 @@
+import p5 from '../core/main';
+import * as constants from '../core/constants';
+import { checkWebGLCapabilities } from './p5.Texture';
+
+class FramebufferCamera extends p5.Camera {
+  constructor(fbo) {
+    super(fbo.target._renderer);
+    this.fbo = fbo;
+  }
+
+  _computeCameraDefaultSettings() {
+    super._computeCameraDefaultSettings();
+    this.defaultAspectRatio = this.fbo.width / this.fbo.height;
+    this.defaultEyeZ =
+      this.fbo.height / 2.0 / Math.tan(this.defaultCameraFOV / 2.0);
+    this.defaultCameraNear = this.defaultEyeZ * 0.1;
+    this.defaultCameraFar = this.defaultEyeZ * 10;
+  }
+
+  resize() {
+    // If we're using the default camera, update the aspect ratio
+    if (this.cameraType === 'default') {
+      this._computeCameraDefaultSettings();
+      this._setDefaultCamera();
+    } else {
+      this.perspective(
+        this.cameraFOV,
+        this.fbo.width / this.fbo.height
+      );
+    }
+  }
+}
+
+p5.FramebufferCamera = FramebufferCamera;
+
+class FramebufferTexture {
+  constructor(framebuffer, property) {
+    this.framebuffer = framebuffer;
+    this.property = property;
+  }
+
+  get width() {
+    return this.framebuffer.width * this.framebuffer.density;
+  }
+
+  get height() {
+    return this.framebuffer.height * this.framebuffer.density;
+  }
+
+  rawTexture() {
+    return this.framebuffer[this.property];
+  }
+}
+
+p5.FramebufferTexture = FramebufferTexture;
+
+class Framebuffer {
+  /**
+   * @class p5.Framebuffer
+   * @constructor
+   * @param {p5.Graphics|p5} target A p5 global instance or p5.Graphics
+   * @param {Object} [settings] A settings object
+   */
+  constructor(target, settings = {}) {
+    this.target = target;
+    this.target._renderer.framebuffers.add(this);
+
+    this.format = settings.format || constants.UNSIGNED_BYTE;
+    this.channels = settings.channels || (
+      target._renderer._pInst._glAttributes.alpha
+        ? constants.RGBA
+        : constants.RGB
+    );
+    this.depthFormat = settings.depthFormat || (
```

Comment:
For completeness should we include the half_float format for depth textures as well?

Reply:
I tried adding it at first but it looks like there just isn't a combination of settings that gives us HALF_FLOAT for depth. I'm not sure how comprehensive this is, but the [third table on this page](https://webgl2fundamentals.org/webgl/lessons/webgl-data-textures.html) seems to confirm that there's just normal float depth component formats:

<img width="409" alt="image" src="https://user-images.githubusercontent.com/5315059/226143947-2fb9d021-662a-4963-822a-3d7c7e5ac950.png">

Reply:
Oh, in that case nevermind! I was also wondering how common uint8 is as a depth format. It seems like it would only give you a very narrow band of values in your depth buffer without much precision.

Reply:
I think that channel is just used for the stencil, so that you could maybe mask out different material classes or something, and you'd maybe only have a few different material types in that case. We might want to look into depth stencil support at some point, but I'm leaving it out for now to have less settings combinations to test haha 😅

Reply:
Ah that would make more sense! Yeah stencil support seems maybe more advanced than we need at this point